Add `type` field

gramakri opened the pull request:
Fixes #10

gramakri committed:
Parse type field

Fixes #10

gramakri commented:
Err, doesn't parse correctly. Will debug this later today

sindresorhus commented:
Needs to be documented too, in the readme and index.d.ts: https://github.com/sindresorhus/df/blob/master/index.d.ts

gramakri commented:
This is turning to be lot more complicated than I expected. I wish df has some formatting output option. This is because the Type field is left-aligned output and 1024-blocks is right-aligned. This means that we cannot find the "boundary" by just looking at the header.

gramakri commented:
I gave your other module `parse-columns` a shot but it has the same problem parsing the columns

gramakri commented:
@sindresorhus I changed the regexp to specify the column names directly. It makes the regexp simpler to parse the header. I fixed the docs and td as well.

sindresorhus commented:
Thanks :)

sindresorhus commented:
I ran the tests locally now and it's failing on macOS:

```
  3 tests failed

  df.file()

  /Users/sindresorhus/dev/oss/df/test.js:26

   25:
   26:   t.true(data.filesystem.length > 0);
   27:   t.true(data.type.length > 0);

  Rejected promise returned by test. Reason:

  TypeError {
    message: 'Cannot read property \'filesystem\' of undefined',
  }



  df()

  /Users/sindresorhus/dev/oss/df/node_modules/execa/lib/error.js:58

  Rejected promise returned by test. Reason:

  Error {
    all: `df: option requires an argument -- T␊
    usage: df [-b | -H | -h | -k | -m | -g | -P] [-ailn] [-T type] [-t] [filesystem ...]`,
    command: 'df -kPT',
    exitCode: 64,
    exitCodeName: 'EHOSTDOWN',
    failed: true,
    isCanceled: false,
    killed: false,
    signal: undefined,
    stderr: `df: option requires an argument -- T␊
    usage: df [-b | -H | -h | -k | -m | -g | -P] [-ailn] [-T type] [-t] [filesystem ...]`,
    stdout: '',
    timedOut: false,
    message: 'Command failed with exit code 64 (EHOSTDOWN): df -kPT',
  }

  makeError (node_modules/execa/lib/error.js:58:11)
  handlePromise (node_modules/execa/index.js:257:18)



  df.fs()

  /Users/sindresorhus/dev/oss/df/node_modules/execa/lib/error.js:58

  Rejected promise returned by test. Reason:

  Error {
    all: `df: option requires an argument -- T␊
    usage: df [-b | -H | -h | -k | -m | -g | -P] [-ailn] [-T type] [-t] [filesystem ...]`,
    command: 'df -kPT',
    exitCode: 64,
    exitCodeName: 'EHOSTDOWN',
    failed: true,
    isCanceled: false,
    killed: false,
    signal: undefined,
    stderr: `df: option requires an argument -- T␊
    usage: df [-b | -H | -h | -k | -m | -g | -P] [-ailn] [-T type] [-t] [filesystem ...]`,
    stdout: '',
    timedOut: false,
    message: 'Command failed with exit code 64 (EHOSTDOWN): df -kPT',
  }

  makeError (node_modules/execa/lib/error.js:58:11)
  handlePromise (node_modules/execa/index.js:257:18)

npm ERR! Test failed.  See above for more details.
```

gramakri commented:
Oh, it looks like maybe this option is not supported on mac? Let me see if I can get hold of a Mac (My desktop is linux..)

kevva commented:
> Oh, it looks like maybe this option is not supported on mac?

It's not. You can use `mount` to get the file system type like below where `apfs` is the type.

```
$ mount
/dev/disk1s1 on / (apfs, local, journaled)
```

sindresorhus commented:
Moved to https://github.com/sindresorhus/df/issues/15
